src/unit_tests/variant/variant_cast_test.cpp

```diff
@@ -0,0 +1,241 @@
+*   the rights to use, copy, modify, merge, publish, distribute, sublicense,        *
+*   and/or sell copies of the Software, and to permit persons to whom the           *
+*   Software is furnished to do so, subject to the following conditions:            *
+*                                                                                   *
+*   The above copyright notice and this permission notice shall be included in      *
+*   all copies or substantial portions of the Software.                             *
+*                                                                                   *
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      *
+*   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        *
+*   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     *
+*   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          *
+*   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   *
+*   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   *
+*   SOFTWARE.                                                                       *
+*                                                                                   *
+*************************************************************************************/
+
+#include <catch/catch.hpp>
+#include <iostream>
+#include <rttr/type>
+
+using namespace rttr;
+
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(variant*)", "[variant]")
+{
+    SECTION("empty cast")
+    {
+        variant var;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value == nullptr);
+    }
+
+    SECTION("valid cast")
+    {
+        variant var = 12;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value != nullptr);
+
+        static_assert(std::is_same<decltype(value), int*>::value, "Must be integer pointer type!");
+
+        REQUIRE(value != nullptr);
+        CHECK(*value == 12);
+    }
+
+    SECTION("valid cast - pointer")
+    {
+        int obj = 12;
+        variant var = &obj;
+
+        auto value = variant_cast<int*>(&var);
+        REQUIRE(value != nullptr);
+        CHECK(**value == 12);
+    }
+
+    SECTION("invalid cast")
+    {
+        variant var = 12;
+
+        auto value = variant_cast<bool>(&var);
+        CHECK(value == nullptr); // nullptr, because not of type "int"
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE("variant - variant_cast<T>(const variant*)", "[variant]")
+{
+    SECTION("empty cast")
+    {
+        const variant var;
+
+        auto value = variant_cast<int>(&var);
+
+        CHECK(value == nullptr);
+    }
+
+    SECTION("valid cast")
+    {
+        const variant var = 12;
+
+        auto value = variant_cast<int>(&var);
```

Comment:
so the const semantic  in `variant_cast<...>` is neglected. meaning we dont write `variant_cast<const int>`.
Wouldnt that be better? I see the benefits of the current implementation. I would leave it like that

Reply:
@gabyx 
What would be better?
You cannot have a `const-by-value` inside a variant. However, you can have a `const int*`